packages/app-check/src/definitions.ts

```diff
@@ -101,7 +102,29 @@ export interface InitializeOptions {
    */
   isTokenAutoRefreshEnabled?: boolean;
   /**
-   * The reCAPTCHA v3 site key (public key).
+   * Set used app check provider for Web.
+   * 
+   * Only available for Web.
+   *
+   * Read more: https://firebase.google.com/docs/app-check/web/custom-provider
+   * 
+   * @since TBA
+   * @default ReCaptchaV3Provider
+   */
+  provider?: 'ReCaptchaV3Provider' | 'ReCaptchaEnterpriseProvider' | 'CustomProvider';
```

Comment:
```suggestion
  provider?: Provider;
```

Let's create an enum for this:

```ts
/**
 * @since 7.0.0
 */
export enum Provider {
  /**
   * @since 7.0.0
   */
  CustomProvider = 'CustomProvider',
  /**
   * @since 7.0.0
   */
  ReCaptchaEnterpriseProvider = 'ReCaptchaEnterpriseProvider',
  /**
   * @since 7.0.0
   */
  ReCaptchaV3Provider = 'ReCaptchaV3Provider'
}
```